src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -261,18 +262,36 @@ std::vector<BlockMetadata> RelationalExpression<Comparison>::evaluateImpl(
     }
   }
 
+  // Helper to retrieve the reference `ValueId` from `IdOrLocalVocabEntry`
+  // variant.
+  auto getValueIdFromReferenceValue =
+      [](const IdOrLocalVocabEntry& referenceValue, LocalVocab& vocab) {
+        return std::visit(
+            ad_utility::OverloadCallOperator{
+                [](const ValueId& referenceId) -> ValueId {
```

Comment:
I think sonarcloud doesn't like the redundant explicit return types, and I think you also don't need them here.